src/XRP/model/xrp-transaction.ts

```diff
@@ -67,39 +101,66 @@ export default class XrpTransaction {
         .map((signer) => XrpSigner.from(signer, xrplNetwork))
     }
 
-    const sourceTag = transaction.getSourceTag()?.getValue()
+    const signingPublicKey = transaction.getSigningPublicKey()?.getValue_asB64()
+    if (signingPublicKey == undefined) {
```

Comment:
`===`

Reply:
https://howtodoinjava.com/typescript/equals-vs-strict-equals/  interesting, don't think I knew this.  What's the risk of using `==` specifically in the context of a comparison with `undefined`?  will it give erroneous results?  I'm wondering if there are other places where we need to update this.

Also, is there a typescript operator that compares not just for same type and same value, but same object?  kinda the way `===` essentially compares memory addresses in... ? java I think?

Reply:
> https://howtodoinjava.com/typescript/equals-vs-strict-equals/ interesting, don't think I knew this. What's the risk of using == specifically in the context of a comparison with undefined? will it give erroneous results? I'm wondering if there are other places where we need to update this.

in this case, probably nothing - the only thing that `== undefined` is `null`. however, `==` does have some interesting behaviors: https://www.ecma-international.org/ecma-262/5.1/#sec-11.9.3

as such it's pretty common to turn on linting (https://eslint.org/docs/rules/eqeqeq) and have only one equality operator

> Also, is there a typescript operator that compares not just for same type and same value, but same object? kinda the way === essentially compares memory addresses in... ? java I think?

`===` compares memory addresses!! this is really important to how libraries like `React` and other functional and functional-y libraries work, particularly where places where changes need to be listened for

```
const obj1 = {}
const obj2 = {}
obj1 === obj2
> false
const obj1Ref = obj1
obj1 === obj1Ref
> true